portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -59,8 +59,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
+    // Get number of comments to display
+    int numCommentsToDisplay;
+    try {
+      numCommentsToDisplay = Integer.parseInt(request.getParameter("limit"));
```

Comment:
This limit could negative so you also want to handle negative numbers. 

Andrew has mentioned in other's PRs:

> It's also a good idea in general to set a maximum number of comments.
> Right now I can crash your server by requesting more comments than your server can hold in memory (assuming there are actually that many in the database)